### PR TITLE
Move all build logic and dependencies installs to Makefile

### PR DESCRIPTION
**Pull request by adamdubiel**
travis.ci configuration uses make commands to be able to replicate CI builds
on local machine

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,26 +3,7 @@ language: go
 go:
   - 1.7
 before_install:
-  - go get golang.org/x/tools/cmd/cover
-  - go get github.com/axw/gocov/gocov
```
Comment (adamdubiel) [inline .travis.yml]: i removed those "magic" installs as they don't seem to be used anymore

File: Makefile [hunk 2]
```diff
@@ -1,40 +1,62 @@
...
+
+TESTARGS ?= -race
+
+CURRENTDIR = $(shell pwd)
+SOURCEDIR = $(CURRENTDIR)
 SOURCES := $(shell find $(SOURCEDIR) -name '*.go')
 APP_SOURCES := $(shell find $(SOURCEDIR) -name '*.go' -not -path '$(SOURCEDIR)/vendor/*')
-VERSION=$(shell cat .goxc.json | python -c "import json,sys;obj=json.load(sys.stdin);print obj['PackageVersion'];")
+
+PATH := $(CURRENTDIR)/bin:$(PATH)
```
Comment (adamdubiel) [inline Makefile]: setting this path once for all tests, thanks to running builds with consul on travis we got +10% to coverage for free ;)

File: Makefile [hunk 3]
```diff
@@ -1,40 +1,62 @@
...
+
+COVERAGE_TARGETS = $(PACKAGES)
 
 all: build
 
 deps:
 	@./install_consul.sh
 
-build: deps test
+build-deps: deps format test check
```
Comment (adamdubiel) [inline Makefile]: build runs `goimports` before starting build

**Comment by janisz** [comment]
There is a problem with tarvis build. I restarted it but it failed again.

**Comment by adamdubiel** [comment]
...and now it passed. Either we get back to not running `consul` tests with Consul on CI right now, or we wait to see how reliable this tests are in longer run, what do you think?

File: Makefile [hunk 4]
```diff
@@ -1,40 +1,62 @@
...
+	go test -coverprofile=coverage/$(shell basename $@).coverprofile $(TESTARGS) $@
 
-gometalint-exists:
+check-deps: deps
 	@which gometalinter > /dev/null || \
-	(echo >&2 "gometalinter must be installed on the system. See https://github.com/alecthomas/gometalinter")
+        (go get github.com/alecthomas/gometalinter && gometalinter --install)
 
-check: gometalint-exists $(SOURCES)
+check: check-deps $(SOURCES)
```
Comment (janisz) [inline Makefile]: I think we should follow [standard targets](https://www.gnu.org/software/make/manual/make.html#index-check-_0028standard-target_0029) so check should do style checking and testing.

File: Makefile [hunk 5]
```diff
@@ -1,40 +1,62 @@
...
-	PATH=$(CURRENT_DIR)/bin:$(PATH) go test $(PACKAGES) $(TESTARGS)
-	go vet $(PACKAGES)
+	go test $(PACKAGES) $(TESTARGS)
+
+coverage-deps: deps
+	@mkdir -p $(COVERAGEDIR)
+	@which gover > /dev/null || \
+        (go get github.com/modocache/gover)
+
+test-coverage: coverage-deps $(SOURCES) $(COVERAGE_TARGETS)
```
Comment (janisz) [inline Makefile]: I'd prefer to keep one test target. I think `test-coverage` looks better then current `test`, the output is cleaner and gives information about coverage.

Comment (adamdubiel) [inline Makefile]: okay, that's reasonable, it doesn't seem to be slower

File: Makefile (hunk 4 above)
Comment (adamdubiel) [inline Makefile]: ack

**Comment by janisz** [comment]
I think we should merge it. I think build hang could be not related to changes you made.

**Comment by adamdubiel** [comment]
I know its not related to changes, but we might not be happy with builds failing randomly because of consul.

**Comment by adamdubiel** [comment]
@janisz fixed, now we have 2 targets: `test` (testing with coverage) and `check` (test + metalinter)

**Pull request merged**